runtime/service.go

```diff
@@ -318,6 +323,9 @@ func (s *service) StartShim(shimCtx context.Context, containerID, containerdBina
 	_, err = fcControlClient.CreateVM(shimCtx, &proto.CreateVMRequest{
 		VMID:                  s.vmID,
 		ExitAfterAllTasksGone: exitAfterAllTasksGone,
+		// if containerCount is zero, that signifies that the VM has already been
```

Comment:
It doesn't signify that the VM has already been started.  If the VM has already, definitely been started, we wouldn't need to call CreateVM either.  This code handles the case where:

* A VM ID has been specified, and
* CreateVM was *not* called

and ends up creating a VM that can run exactly 1 container.

The comment should reflect this.

I'm wondering if we should just specify 1 here regardless, since the 0 that we're specifying gets overridden to 1 inside the CreateVM function body anyway and a duplicate call to CreateVM for a VM that has already been created should be ignored.

Reply:
I think we should leave it as 0 as 0 indicates "I'm not specifying anything, give me the default ContainerCount". For this codepath (vmID specified but CreateVM not called yet), we don't necessarily need a VM with only 1 container, we just want a VM with default values.

Currently, 1 is the default ContainerCount but it doesn't necessarily have to be; it could just as easily be the maximum number of drives a VM allows or any other value really. Leaving the value as 0 here allows us to avoid harcoding defaults in multiple independent locations. We instead just set the default once in `buildVMConfiguration`, where we also fill in all other defaults for other VM parameters.